### ISSUE TITLE
Do not emit unknown property warning for configured _instanceProps

### ISSUE THREAD

**Issue by web-padawan** [Polymer/polymer-linter#67]
In my [brainy-table](https://github.com/OWOX/brainy-table) element I configure instance properties like this:
```js
    created: function() {
      this._instanceProps = {
        column: true,
        expanded: true,
        index: true,
        item: true
      };
    },
```
So user shouldn't get errors like this:
```
 [databind-with-unknown-property] - item is not declared or used more than once.
```

Is it a way to perform deeper analysis than [this one](https://github.com/Polymer/polymer-linter/blob/master/src/polymer/databind-with-unknown-property.ts#L58) which only respects `dom-repeat` templates?

**Comment by rictic** [comment]
I see. What do you think about this plan:

- when we notice a `<template>` element we try to discover its owner:
  - if the template tag has an `is` attribute, that identifies the tagname of its owner
  - otherwise its nearest ancestor in the dom that is known to be a templatizer element is its owner

Then the question is how best to document that an element is a templatizer element, and how to document its properties. We could invent a method on the element like `documentTemplatizerProperties` which must return a polymer properties object, like:

```javascript
documentTemplatizerProperties: function() {
  return {
    /** Docs for column. */
    column: Number,
    /** Docs for expanded. */
    expanded: Boolean,
    /** Docs for index. */
    index: Number,
    /** Docs for item. */
    item: Object,
  };
}
```

That way we could not only not warn for these properties, but also give docs for them in an IDE context.

**Comment by web-padawan** [comment]
Adding separate method looks not very DRY, but it might be the only solution appropriate for both versions, because of making `instanceProps` [public](https://www.polymer-project.org/2.0/docs/api/namespaces/Polymer.Templatize) in 2.0. 

BTW, this should be relevant for [iron-list](https://github.com/PolymerElements/iron-list/blob/952dce24a3571798c8c8725bae9f8a800eb8e954/iron-list.html#L1066) too, as it configures `_instanceProps` in the way similar to `dom-repeat`: they both use `as` and `indexAs`.
